added the files from wdl-common-fef058b879d04c15c3da2626b320afdd8ace6…

shadizaheri opened the pull request:
…c2e to wdl-common folder

williamrowell commented:
If you're looking for releases without submodules, the `*-dockstore` tags are probably want you want.  We remove the submodules and directly add these files to the branch for release.  For example: https://github.com/PacificBiosciences/HiFi-human-WGS-WDL/tree/v1.1.0-dockstore

These are the default releases on Dockstore, e.g. https://dockstore.org/workflows/github.com/PacificBiosciences/HiFi-human-WGS-WDL/HiFi-human-WGS-WDL:v1.1.0-dockstore?tab=info

shadizaheri commented:
Hi Billy, 
Thank you for pointing out the *-dockstore tags for submodule-free releases. 

I was working to import your sample_analysis WDL into Terra. Due to access restrictions, I couldn't import your `sample_analysis` wdl directly from your released tags. Considering that submodules were missing from your main branch, I [forked your repository](https://github.com/shadizaheri/HiFi-human-WGS-WDL), added the submodules from the `wdl-common` at commit fef058b to a new `wdl_common` directory, and updated references accordingly.

([here](https://dockstore.org/my-workflows/github.com/shadizaheri/HiFi-human-WGS-WDL/HiFi_human_WGS_sample_analysis) is an example of how I imported your sample_analysis wdl to the dockstore)

@williamrowell I have a question about your main workflow, please. Is it feasible to initially run your [main WDL](https://dockstore.org/workflows/github.com/PacificBiosciences/HiFi-human-WGS-WDL/HiFi-human-WGS-WDL:v1.1.0-dockstore?tab=info) on individual samples separately and subsequently use the same WDL for joint calling using the outputs from the individual results? Would this scenario require resubmitting all samples together as a cohort for joint calling, or can they be processed sequentially as decisions for joint calling are made?

williamrowell commented:
Last month I spoke with Kiran and Mark about this workflow.  I'm in the middle of writing v2, which is restructured to fit better for individual samples in Terra.  It runs end to end now, and I just need to stress test it more before merging.

Running on individual samples produces a gVCF for small variant calls that can be used later for joint-calling with glnexus, but there aren't any gVCF equivalents for the other variant callers.

We joint call with pbsv in this workflow as well, but the intermediate files, `*.svsig.gz`, are large and the format isn't completely stable between versions, so we decided not to keep these intermediates.  We have plans to replace pbsv with a new structural variant caller.

Tools exist to merge SV VCFs from multiple samples (not joint-calling, just merging), but I don't have any I can whole-heartedly recommend at the moment.
